ext/dba/libinifile/inifile.c

```diff
@@ -471,78 +457,76 @@ static int inifile_delete_replace_append(inifile *dba, const key_type *key, cons
-					ret = FAILURE;
-				}
-			}
+	fp_tmp = php_stream_temp_create(0, 64 * 1024);
+	if (!fp_tmp) {
+		php_error_docref(NULL, E_WARNING, "Could not create temporary stream");
+		goto end;
+	}
+
+	php_stream_seek(dba->fp, 0, SEEK_END);
+	if (pos_grp_next != (size_t)php_stream_tell(dba->fp)) {
+		php_stream_seek(dba->fp, pos_grp_next, SEEK_SET);
+		if (SUCCESS != php_stream_copy_to_stream_ex(dba->fp, fp_tmp, PHP_STREAM_COPY_ALL, NULL)) {
+			php_error_docref(NULL, E_WARNING, "Could not copy remainder to temporary stream");
+			goto end;
 		}
 	}
 
 	/* 5 */
-	if (ret == SUCCESS) {
-		if (!value || (key->name && strlen(key->name))) {
-			ret = inifile_truncate(dba, append ? pos_grp_next : pos_grp_start); /* writes error on fail */
+	if (!value || (key->name && strlen(key->name))) {
+		bool has_succeeded = inifile_truncate(dba, is_append ? pos_grp_next : pos_grp_start);
+		if (!has_succeeded) {
+			/* Warning is already emited by inifile_truncate() on failure */
+			goto end;
 		}
 	}
 
-	if (ret == SUCCESS) {
-		if (key->name && strlen(key->name)) {
-			/* 6 */
-			if (!append && ini_tmp) {
-				ret = inifile_filter(dba, ini_tmp, key, found);
-			}
-
-			/* 7 */
-			/* important: do not query ret==SUCCESS again: inifile_filter might fail but
-			 * however next operation must be done.
-			 */
-			if (value) {
-				if (pos_grp_start == pos_grp_next && key->group && strlen(key->group)) {
-					php_stream_printf(dba->fp, "[%s]\n", key->group);
-				}
-				php_stream_printf(dba->fp, "%s=%s\n", key->name, value->value ? value->value : "");
-			}
+	ret = SUCCESS;
+	if (key->name && strlen(key->name)) {
+		/* 6 */
+		if (!is_append && ini_tmp) {
+			ret = inifile_filter(dba, ini_tmp, key, found);
 		}
 
-		/* 8 */
+		/* 7 */
 		/* important: do not query ret==SUCCESS again: inifile_filter might fail but
 		 * however next operation must be done.
 		 */
-		if (fp_tmp && php_stream_tell(fp_tmp)) {
-			php_stream_seek(fp_tmp, 0, SEEK_SET);
-			php_stream_seek(dba->fp, 0, SEEK_END);
-			if (SUCCESS != php_stream_copy_to_stream_ex(fp_tmp, dba->fp, PHP_STREAM_COPY_ALL, NULL)) {
-				zend_throw_error(NULL, "Could not copy from temporary stream - ini file truncated");
-				ret = FAILURE;
+		if (value) {
+			if (pos_grp_start == pos_grp_next && key->group && strlen(key->group)) {
+				php_stream_printf(dba->fp, "[%s]\n", key->group);
 			}
+			php_stream_printf(dba->fp, "%s=%s\n", key->name, value->value ? value->value : "");
 		}
 	}
 
+	/* 8 */
+	/* important: do not query ret==SUCCESS again: inifile_filter might fail but
+	 * however next operation must be done.
+	 */
+	if (fp_tmp && php_stream_tell(fp_tmp)) {
+		php_stream_seek(fp_tmp, 0, SEEK_SET);
+		php_stream_seek(dba->fp, 0, SEEK_END);
+		if (SUCCESS != php_stream_copy_to_stream_ex(fp_tmp, dba->fp, PHP_STREAM_COPY_ALL, NULL)) {
+			zend_throw_error(NULL, "Could not copy from temporary stream - ini file truncated");
+			ret = FAILURE;
+		}
+	}
+
+	end:
```

Comment:
I generally prefer putting it unindented, Tim prefers a single space of indent, and you have have a third style :P
we should make some rules for this. In any case, the label is easy to miss imo

Reply:
I'm fine to have it unindented, but yes a rule would be good.